qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1112,13 +1112,17 @@ def substitute_node(self, node, op, inplace=False):
                     op.num_qubits, op.num_clbits))
 
         if inplace:
+            save_condition = node.condition
             node.op = op
             node.name = op.name
+            node.condition = save_condition
```

Comment:
```suggestion
            node.op.condition = save_condition
```
A suggestion for for clarity. (For someone who didn't know that `node.condition` was a `@property` pointing into `node.op`, this would read like `node.condition = node.condition`. Maybe this is sufficient reason to remove `node.condition`, but we can discuss that separately.)

Reply:
That's fine. As to removing node.condition, there are about 30 uses in the code (though there may be more that use something other than 'node'). My assumption is that in those cases there is not always a prior test of op being an attribute of node. So that would have to be done if it's changed everywhere to node.op.condition.

Reply:
> My assumption is that in those cases there is not always a prior test of op being an attribute of node.

Good point. This is worth looking into. My guess would be if there's code that's accessing a condition of a node that's not an op node, we would want to raise (similar to how we do for setting) but I'd be curious to see how its being used in practice.

Reply:
One quick way to check is on the `@property` for condition in `DAGNode`, if when you check that it's an op, instead of returning None if it's not, you raise. If you then run the circuit tests, you get 450 errors. A lot of those may be single tests with many multiple inputs, but you can see that `node.condition` is used rather loosely. 

I do raise on the setter for condition and that doesn't cause any problems.

Reply:
That's a good idea. Looking quickly, it looks like most of those failures originate from `DAGNode.__eq__` ( https://github.com/Qiskit/qiskit-terra/blob/8f45a79/qiskit/dagcircuit/dagnode.py#L142 ) which is used heavily in checking for circuit equality in tests. If this is the only problematic usage, we should be able to resolve the test failures there.

Reply:
I'll do a run-through with the tests and see if it can be simply fixed, and if so, I'll add the op check to the `@property`.

If that works, then as a second phase (maybe a new issue) it would probably be ok to replace `node.condition` with `node.op.condition` throughout.

Reply:
Sounds good, thanks! (and agree removing `node.condition` can be another issue)

Reply:
@kdk Good call on `DAGNode.__eq__`. Fixing that eliminated all the test errors. Should make a later remove of `node.condition` straightforward as well. All changes done in https://github.com/Qiskit/qiskit-terra/pull/6040/commits/d0ad7a3f3441ad5a2620d04459f5ff2eeedf0338